app/domain/authentication/authn_jwt/identity_provider_factory.rb

```diff
@@ -0,0 +1,23 @@
+module Authentication
+  module AuthnJwt
+    # Factory for jwt identity providers.
+    # If Identity variable is configured factory return the decoded_token_provider
+    # If the Identity variable is not configured and there is account field in url factory returns url provider
+    # If the above conditions are not met exception is raised
+    class IdentityProviderFactory
+      PROVIDERS = {
+        "from_url_provider" => Authentication::AuthnJwt::IdentityFromUrlProvider,
+        "from_decoded_token_provider" => Authentication::AuthnJwt::IdentityFromDecodedTokenProvider
+      }
+
+      def self.relevant_id_provider(authentication_parameters)
+        if PROVIDERS["from_decoded_token_provider"].new(authentication_parameters).identity_available?
```

Comment:
Please avoid of creating the objects twice

Reply:
Avoided